File: nas_spec/code_samples/Java/marketplace@entities@{id}/put.java
// For more information please refer to https://github.com/checkout/checkout-sdk-java
import com.checkout.CheckoutApiException;
import com.checkout.CheckoutArgumentException;
import com.checkout.CheckoutAuthorizationException;
import com.checkout.CheckoutSdk;
import com.checkout.Environment;
import com.checkout.common.Address;
import com.checkout.common.CountryCode;
import com.checkout.four.CheckoutApi;
import com.checkout.marketplace.ContactDetails;
import com.checkout.marketplace.DateOfBirth;
import com.checkout.marketplace.Document;
import com.checkout.marketplace.DocumentType;
import com.checkout.marketplace.Identification;
import com.checkout.marketplace.Individual;
import com.checkout.marketplace.OnboardEntityRequest;
import com.checkout.marketplace.OnboardEntityResponse;
import com.checkout.marketplace.Phone;
import com.checkout.marketplace.Profile;

CheckoutApi api = CheckoutSdk.fourSdk()
    .oAuth()
    .clientCredentials("client_id", "client_secret")
    .environment(Environment.SANDBOX) // or Environment.PRODUCTION
    .scopes(FourOAuthScope.MARKETPLACE) // more scopes available
    .build();

OnboardEntityRequest request = OnboardEntityRequest.builder()
    .reference("reference")
    .contactDetails(ContactDetails.builder()
        .phone(Phone.builder().number("415 555 2671").build())
        .build())
    .profile(Profile.builder()
        .urls(Collections.singletonList("https://docs.checkout.com"))
        .mccs(Collections.singletonList("0742"))
        .build())
    .individual(Individual.builder()
        .firstName("FirstName")
        .lastName("LastName")
        .tradingName("Trading")
        .registeredAddress(Address.builder()
            .addressLine1("Checkout.com")
            .addressLine1("90 Tottenham Court Road")
            .city("London")
            .state("London")
            .zip("WIT 4TJ")
            .country(CountryCode.GB)
            .build())
        .nationalTaxId("TAX123456")
        .dateOfBirth(DateOfBirth.builder()
            .day(5)
            .month(6)
            .year(1995)
            .build())
        .identification(Identification.builder()
            .nationalIdNumber("AB123456C")
            .document(Document.builder()
                    .back("number")
                    .front("number")
                    .type(DocumentType.PASSPORT)
                    .build())
            .build())
        .build())
    .build();

try {
    OnboardEntityResponse response = api.marketplaceClient().updateEntity(request, "entity_id").get();
} catch (CheckoutApiException e) {
    // API error
    String requestId = e.getRequestId();
    int statusCode = e.getHttpStatusCode();
    Map<String, Object> errorDetails = e.getErrorDetails();
} catch (CheckoutArgumentException e) {
    // Bad arguments
} catch (CheckoutAuthorizationException e) {
    // Invalid authorization
}

File: nas_spec/code_samples/C#/marketplace@entities@{id}/put.cs
// For more information please refer to https://github.com/checkout/checkout-sdk-net
using Checkout.Common;
using Checkout.Marketplace;
using Phone = Checkout.Marketplace.Phone;

Four.ICheckoutApi api = CheckoutSdk.FourSdk().OAuth()
    .ClientCredentials("client_id", "client_secret")
    .Scopes(FourOAuthScope.Marketplace)
    .Environment(Environment.Sandbox)
    .FilesEnvironment(Environment.Sandbox)
    .Build();

OnboardEntityRequest request = new OnboardEntityRequest
{
    Reference = "reference",
    ContactDetails = new ContactDetails()
    {
        Phone = new Phone()
        {
            Number = "2345678910"
        }
    },
    Profile = new Profile()
    {
        Urls = new List<string>()
        {
            "https://www.superheroexample.com"
        },
        Mccs = new List<string>()
        {
            "5669"
        }
    },
    Individual = new Individual()
    {
        FirstName = "FirstName",
        LastName = "LastName",
        TradingName = "John's Super Hero Masks",
        RegisteredAddress = new Address()
        {
            AddressLine1 = "Checkout.com",
            AddressLine2 = "90 Tottenham Court Road",
            City = "London",
            State = "London",
            Zip = "W1T 4TJ",
            Country = CountryCode.GB
        },
        NationalTaxId = "TAX123456",
        DateOfBirth = new DateOfBirth()
        {
            Day = 5,
            Month = 6,
            Year = 1995
        },
        Identification = new Identification()
        {
            NationalIdNumber = "AB123456C"
        }
    }
};

try
{
    OnboardEntityResponse response = api.MarketplaceClient().UpdateEntity("entity_id", request).Result;
}
catch (CheckoutApiException e)
{
    // API error
    string requestId = e.RequestId;
    var statusCode = e.HttpStatusCode;
    IDictionary<string, object> errorDetails = e.ErrorDetails;
}
catch (CheckoutArgumentException e)
{
    // Bad arguments
}
catch (CheckoutAuthorizationException e)
{
    // Invalid authorization
}